### PR TITLE
Added Hardkernel ODROID-XU4 board support

### PR DESCRIPTION
**Pull request by bkrepo**
Added Hardkernel ODROID-XU4 board support with Samsung Exynos5422 SoC.

### CODE REVIEW

File: src/soc/soc.c [hunk 1]
```diff
@@ -23,6 +23,8 @@
 static struct soc_t *socs = NULL;
 
 void soc_register(struct soc_t **soc, char *brand, char *type) {
+	int i;
```
Comment (CurlyMoo) [inline src/soc/soc.c:26]: Can you always keep variables initialized?

File: src/soc/samsung/exynos5422.c [hunk 2]
```diff
@@ -0,0 +1,542 @@
...
+	{ "GPIO_Z1", 4, { 0x000, 4  }, { 0x004, 1 }, FUNCTION_DIGITAL | FUNCTION_INTERRUPT, PINMODE_NOT_SET, 0 },
+	{ "GPIO_Z2", 4, { 0x000, 8  }, { 0x004, 2 }, FUNCTION_DIGITAL | FUNCTION_INTERRUPT, PINMODE_NOT_SET, 0 },
+	{ "GPIO_Z3", 4, { 0x000, 12 }, { 0x004, 3 }, FUNCTION_DIGITAL | FUNCTION_INTERRUPT, PINMODE_NOT_SET, 0 },
+	{ "GPIO_Z4", 4, { 0x000, 16 }, { 0x004, 4 }, FUNCTION_DIGITAL | FUNCTION_INTERRUPT, PINMODE_NOT_SET, 0 },
+	{ "GPIO_Z5", 4, { 0x000, 20 }, { 0x004, 5 }, FUNCTION_DIGITAL | FUNCTION_INTERRUPT, PINMODE_NOT_SET, 0 },
+	{ "GPIO_Z6", 4, { 0x000, 24 }, { 0x004, 6 }, FUNCTION_DIGITAL | FUNCTION_INTERRUPT, PINMODE_NOT_SET, 0 },
+};
+
+static int exynos5422Setup(void) {
+	int i;
```
Comment (CurlyMoo) [inline src/soc/samsung/exynos5422.c:281]: Also here.

File: src/soc/soc.c (hunk 1 above)
Comment (bkrepo) [inline src/soc/soc.c:26]: I will change the variable definition with initial value. 
Do you have any more comments?

Comment (CurlyMoo) [inline src/soc/soc.c:26]: No, i believe not.

**Pull request closed without merging**